### PR TITLE
#2014 Support k0s

### LINKED ISSUE #2014: Support k0s containerd.sock path
For k0s the path is /var/run/k0s/containerd.sock

the current agent error is:
```
2023-10-19T02:38:19.104977Z DEBUG ThreadId(01) mirrord_agent: main -> Initializing mirrord-agent, version 3.72.1.
2023-10-19T02:38:19.105146Z DEBUG ThreadId(01) mirrord_agent: start_iptable_guard -> Initializing iptable-guard.
2023-10-19T02:38:19.105217Z TRACE ThreadId(01) new: mirrord_agent::container_handle: new container=Containerd(ContainerdContainer { container_id: "c8d6b7df532ca7a51ce9870447baa0a90b9d81faa2f069648d97b4ed0c72 03eb" }) watch=Watch { drained_tx: Sender { chan: Tx { inner: Chan { tx: Tx { block_tail: 0x55a66d813450, tail_position: 0 }, semaphore: Semaphore { semaphore: Semaphore { permits: 1 }, bound: 1 }, rx_waker: AtomicWaker, tx_count: 1, rx_fields: "..." } } }, signal_rx: Receiver { shared: Shared { value: RwLock { data: (), poisoned: false, .. }, version: Version(0), is_closed: false, ref_count_rx: 1 }, version: Version(0) } }
2023-10-19T02:38:19.105378Z TRACE ThreadId(01) new: mirrord_agent::container_handle: close time.busy=139µs time.idle=21.6µs container=Containerd(ContainerdContainer { container_id: "c8d6b7df532ca7a51ce9870447baa0a90b9d81faa2f069648d97b4ed0c7203eb" }) watch=Watch { drained_tx: Sender { chan: Tx { inner: Chan { tx: Tx { block_tail: 0x55a66d813450, tail_position: 0 }, semaphore: Semaphore { semaphore: Semaphore { permits: 1 }, bound: 1 }, rx_waker: AtomicWaker, tx_count: 1, rx_fields: "..." } } }, signal_rx: Receiver { shared: Shared { value: RwLock { data: (), poisoned: false, .. }, version: Version(0), is_closed: false, ref_count_rx: 1 }, version: Version(0) } }
2023-10-19T02:38:19.105404Z ERROR ThreadId(01) mirrord_agent: main -> mirrord-agent `start` exiting with error ContainerdSocketNotFound  
```

### PR DESCRIPTION
**Pull request by aarnaud**
Add support for k0s 
Closes #2014

### CODE REVIEW

File: changelog.d/2014.added.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add k0s containerd socket path.
```
Comment (aviramha) [inline changelog.d/2014.added.md:1]: I'd change the wording a bit -
Added k0s support - add k0s containerd socket path.

**Comment by aviramha** [comment]
Thanks! Small nitpick on the changelog and ready to go!

**Comment by aarnaud** [comment]
Done for the changelog

**Review by aviramha — APPROVED** [review]
Thank you! Closes #2014

**Pull request merged**